kotlin/show-device-location-using-indoor-positioning/src/main/java/com/esri/arcgisruntime/sample/showdevicelocationusingindoorpositioning/MainActivity.kt

```diff
@@ -0,0 +1,404 @@
+import com.esri.arcgisruntime.mapping.view.LocationDisplay
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.sample.showdevicelocationusingindoorpositioning.databinding.ActivityMainBinding
+import com.esri.arcgisruntime.security.UserCredential
+import java.text.DecimalFormat
+import java.util.*
+
+class MainActivity : AppCompatActivity(), LocationDataSource.LocationChangedListener,
+    StatusChangedListener {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    private var currentFloor: Int? = null
+
+    // Provides an indoor or outdoor position based on device sensor data (radio, GPS, motion sensors).
+    private var mIndoorsLocationDataSource: IndoorsLocationDataSource? = null
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val progressBar: ProgressBar by lazy {
+        activityMainBinding.progressBar
+    }
+
+    private val textView: TextView by lazy {
+        activityMainBinding.textView
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+        // check for location permissions
+        // if permissions is allowed, the device's current location is shown
+        checkPermissions()
+    }
+
+    /**
+     * Set up the connection between the device and the portal item
+     */
+    private fun connectToPortal() {
+        // set the portal url and the credentials needed
+        val portal = Portal("https://viennardc.maps.arcgis.com/", true).apply {
+            credential = UserCredential("tester_viennardc", "password.testing12345")
+        }
+        val portalItem = PortalItem(portal, "89f88764c29b48218366855d7717d266")
+        setupMap(portalItem)
+    }
+
+    /**
+     * Set the [mapView] using the [portalItem] then invokes [loadTables]
+     */
+    private fun setupMap(portalItem: PortalItem) {
+        mapView.map = ArcGISMap(portalItem)
+        val map = mapView.map
+        map.addDoneLoadingListener {
+            if (mapView.map.loadStatus == LoadStatus.LOADED) {
+                val featureTables = map.tables
+                // check if the portalItem contains featureTables
+                if (featureTables.isNotEmpty()) {
+                    setUpLoadTables(featureTables)
+                } else {
+                    val message = "Map does not contain feature tables"
+                    Toast.makeText(this@MainActivity, message, Toast.LENGTH_SHORT).show()
+                    Log.e(TAG, message)
+                }
+            } else {
+                val error = map.loadError
+                Toast.makeText(this@MainActivity, error.message, Toast.LENGTH_SHORT).show()
```

Comment:
I'd tried to contextualize this as we normally do:
- Write to Log.e also
- "Error loading map: " + map.loadError.message